libbeat/autodiscover/providers/docker/docker.go

```diff
@@ -383,7 +383,7 @@ func (d *Provider) generateHints(event bus.Event) bus.Event {
 		e["ports"] = ports
 	}
 	if labels, err := dockerMeta.GetValue("labels"); err == nil {
-		hints := utils.GenerateHints(labels.(mapstr.M), "", d.config.Prefix)
+		hints, _ := utils.GenerateHints(labels.(mapstr.M), "", d.config.Prefix, []string{})
```

Comment:
didn't know we call `GenerateHints` here. Looking on this use case, wondering if it wouldn't be a better option to introduce additionally smth like `GenerateHintsWithValidation` instead - maybe we should consider it in future for cases when signature is changed

Reply:
Here `GenerateHints` will actually return all hints as incorrect but we just ignore them. I think we should make it more clear. Maybe add a bool (`validate`), which is then checked in GenerateHints

Reply:
FYI: https://github.com/elastic/elastic-agent-autodiscover/pull/89